lib/AST/Decl.cpp

```diff
@@ -983,8 +983,11 @@ ExtensionDecl::takeConformanceLoaderSlow() {
 
 Type ExtensionDecl::getInheritedType(unsigned index) const {
   ASTContext &ctx = getASTContext();
-  return ctx.evaluator(InheritedTypeRequest{const_cast<ExtensionDecl *>(this),
-                                            index});
+  auto result = ctx.evaluator(
+    InheritedTypeRequest{const_cast<ExtensionDecl *>(this), index});
+  if (result)
+    return *result;
+  return Type();
```

Comment:
This is great; exactly the point where the default value should get substituted in.